eq-author/src/components/Reorder/index.js

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useEffect, useRef } from "react";
+import styled, { keyframes, css } from "styled-components";
+import { TransitionGroup } from "react-transition-group";
+import PropTypes from "prop-types";
+
+import getIdForObject from "utils/getIdForObject";
+
+const MOVE_DURATION = 400;
+const UP = "UP";
+const DOWN = "DOWN";
+
+const move = ({ transform, scale }) => keyframes`
+  0% {
+    transform: translateY(0) scale(1);
+  }
+  50% {
+    transform: scale(${scale});
+  }
+  100% {
+    transform: translateY(calc(${transform})) scale(1);
+  }
+`;
+
+export const Segment = styled.div`
+  z-index: ${props => props.movement.zIndex};
+  transform-origin: 50% 50%;
+  animation: ${({ movement }) =>
+    movement.transform !== 0 &&
+    css`
+      ${move(
+        movement
+      )} ${MOVE_DURATION}ms cubic-bezier(0.785, 0.135, 0.150, 0.860) 0s forwards 1;
+    `};
+`;
+
+const startingStyles = items => items.map(() => ({ transform: 0 }));
+
+const Reorder = ({ list, onMove, children, Transition }) => {
+  const OuterWrapper = Transition ? TransitionGroup : React.Fragment;
+  const InnerWrapper = Transition || React.Fragment;
+
+  const [renderedItems, setRenderedItems] = useState(list);
+  const [isTransitioning, setIsTransitioning] = useState(false);
+  const [itemStyles, setItemStyles] = useState(startingStyles(list));
+
+  const hasNewitems = useRef(false);
+  const itemElements = useRef([]);
+  const previtems = useRef(list);
+  const animationTimeout = useRef();
+
+  if (previtems.current !== list) {
+    previtems.current = list;
+    hasNewitems.current = true;
+  }
+
+  if (hasNewitems.current && !isTransitioning) {
+    setRenderedItems(list);
+    setItemStyles(startingStyles(list));
```

Comment:
I thought you weren't supposed to use hooks in an `if`?

Reply:
You can use the functions they give you just not the `use...`